src/main/java/org/folio/circulation/domain/UpdateLoan.java

```diff
@@ -57,4 +69,21 @@ public UpdateLoan(Clients clients,
     return loanPolicy.recall(loanAndRelatedRecords.getLoan())
         .map(loanAndRelatedRecords::withLoan);
   }
+
+  private Result<LoanAndRelatedRecords> sendRecallNotice(
+    LoanAndRelatedRecords loanAndRelatedRecords, DateTime dueDateBeforeRecall) {
+
+    Loan loan = loanAndRelatedRecords.getLoan();
+    if (!loan.getDueDate().equals(dueDateBeforeRecall)) {
```

Comment:
An alternative to this would be to track if the due date had changed in the loan, and as `loan.hasDueDateChanged()` rather than needing to capture state outside of the object (and the chain of calls)

What do you think?

Reply:
Agree, it sounds better. I'll do this change.

Reply:
Tried to save the state inside loan object (about the fact that due date has been changed).  But method `loanRepository#updateLoan()` replaces loan with another object so that this state is lost (and I guess there are other places).

Reply:
I think we have three options:
* Retain the change after storing the loan. 
* Only update the change metadata (based upon the assumption that only that part should have changed)
* Not attempt to refresh the loan after the update

I'd prefer to do one of those and continue with this moving of responsibilities, rather than take the previous approach.

What do you think?

Reply:
Yes, if we replace last call in _UpdateLoan#onRequestCreation()_ from
  `.thenApply(r -> r.map(v -> requestAndRelatedRecords));`
to something like 
`.thenApply(r -> r.map(v -> requestAndRelatedRecords.withRequest(request.withLoan(v.getLoan()))));`

because this is another place were loan object is replaced.
Would you like to have this change?

Reply:
I'm not sure I understand, why does the request need changing during the update of the loan?

Reply:
The reason is that in the first call, loan object is loaded by id:
`return loanRepository.getById(loan.getId()).thenApply(r -> r.map(LoanAndRelatedRecords::new))`

Reply:
Loan object is stored inside request object. Request object is immutable, so to replace loan we need to replace request also.

Reply:
I think I'm failing to follow because the loan was already updated prior to this change.

Don't worry, thank you for trying to explain it to me.

Please make the change if you believe it is necessary to pull the sending of the notice further up into the create request service

Reply:
Move sending request-related notices from `RequestNoticeSender` to `CreateRequestService` and `UpdateRequestService`